#
# Copyright (c) 2026 Oracle and/or its affiliates.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

diff --git a/spec/server.go b/spec/server.go
--- a/spec/server.go
+++ b/spec/server.go
@@ -1,6 +1,7 @@
 package spec
 
 import (
 	"crypto/tls"
+	"errors"
 	"fmt"
 	"net"
@@ -62,6 +63,9 @@ func (server *Server) RunListener(listener net.Listener, tc *ClientTestCase) {
 	for {
 		baseConn, err := listener.Accept()
 		if err != nil {
+			if errors.Is(err, net.ErrClosed) {
+				return
+			}
 			log.Println(err)
 			continue
 		}
